packages/credential-provider-web-identity/src/fromTokenFile.ts

```diff
@@ -1,4 +1,6 @@
-import { CredentialProvider } from "@aws-sdk/types";
+import { getDefaultRoleAssumerWithWebIdentity } from "@aws-sdk/client-sts";
```

Comment:
```suggestion
```
This package depending on the STS Client inevitably introduces the circular dependency: Any Client -> `credential-provider-node` -> `credential-provider-web-identity` -> STS Client -> `credential-provider-node`. This is the main reason of all the effort of exportting `getDefaultRoleAssumerWithWebIdentity` from the STS client. So other clients can depend on it directly instead of depending on the `credential-provider-node` thus introducing the circular dependency.

Basically to get the EKS credentials work, you **shouldn't** touch this credential provider. Adding the `fromTokenFile()` to the default provider chain is enough. As a result, the default provider chain can resolve this credential if non of shared credential file and environmental variable is configured with valid credentials.

On the other hand, if you want to explicitly use the `fromTokenFile()`, users MUST always explicitly import `getDefaultRoleAssumerWithWebIdentity` and supply to the `fromTokenFile()`.